sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -383,8 +452,14 @@ func runCtr(containerdSocket string, namespace string, containerID string, sourc
 		log.G(ctrCtx).WithError(err).Error("failed to get container task exit status")
 		return err
 	}
+
 	log.G(ctrCtx).WithField("code", code).Info("container task exited")
 
+	// Return error if container exists with non-zero status
+	if code != 0 {
+		return errors.New("Container exited with non-zero status")
+	}
+
```

Comment:
If by restarted you mean `kill -9 <host-container-pid>`, and wait for the container to come back up, then yes, that still works:

```sh
ps aux | grep host-ctr
root         872  0.1  0.9 1245748 35696 ?       Ssl  00:20   0:00 /usr/bin/host-ctr run --container-id=control

sudo kill -9 872

systemctl status host-containers@control.service

host-containers@control.service - Host container: control
     Loaded: loaded (/aarch64-bottlerocket-linux-gnu/sys-root/usr/lib/systemd/system/host-containers@.service; enabled; vendor preset: enabled)
     Active: activating (auto-restart) (Result: signal) since Wed 2021-04-07 00:22:21 UTC; 21s ago
    Process: 872 ExecStart=/usr/bin/host-ctr run --container-id=control --source=${CTR_SOURCE} --superpowered=${CTR_SUPERPOWERED} (code=killed, signal=KILL)

# waited for some time
systemctl status host-containers@control.service

host-containers@control.service - Host container: control
     Loaded: loaded (/aarch64-bottlerocket-linux-gnu/sys-root/usr/lib/systemd/system/host-containers@.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2021-04-07 00:23:06 UTC; 5s ago
```